sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ConditionEvaluatesToConstant.cs

```diff
@@ -195,7 +195,15 @@ private static IEnumerable<Location> GetUnreachableLocations(SyntaxNode constant
                 // parent has an unreachable branch.
                 if (constantExpression is ExpressionSyntax expression)
                 {
-                    unreachableExpressions.Add(expression);
+                    bool unreachableCodeIsDetectable = constantValue ?
+                    !expression.Parent.IsKind(SyntaxKind.LogicalAndExpression)

```

Comment:
When looking at the parent expression, we should skip parenthesis, otherwise we will still have FP for unreachable code in this case:
```
            if (unknown && (trueConstant)) // Noncompliant
            {

            }
            else
            { // Secondary is raised here for dead code, but it may not be depending on the unknown value

            }
```

Reply:
good point